src/components/PropertyMap.tsx

```diff
@@ -421,45 +450,86 @@ const PropertyMap: FC<PropertyMapProps> = ({
     if (map) {
       updateFilter();
     }
-  }, [map, appFilter]);
+  }, [map, appFilter, currentStyle]);
 
   const changeCursor = (e: any, cursorType: 'pointer' | 'default') => {
     e.target.getCanvas().style.cursor = cursorType;
   };
 
+  map?.on('load', () => {
+    console.log('Map loaded, checking layers...');
+
+    if (!map.getLayer('vacant_properties_tiles_points')) {
+      map.addLayer(layerStylePoints);
+    }
+    if (!map.getLayer('vacant_properties_tiles_polygons')) {
+      map.addLayer(layerStylePolygon);
+    }
+
+    if (
+      map.getLayer('vacant_properties_tiles_points') &&
+      map.getLayer('vacant_properties_tiles_polygons')
+    ) {
+      console.log('Both layers found, applying filters...');
+
+      const mapFilter = Object.entries(appFilter).reduce(
+        (acc, [property, filterItem]) => {
+          if (filterItem.values.length) {
+            const thisFilterGroup: any = ['any'];
+            filterItem.values.forEach((item) => {
+              if (filterItem.useIndexOfFilter) {
+                thisFilterGroup.push([
+                  '>=',
+                  ['index-of', item, ['get', property]],
+                  0,
+                ]);
+              } else {
+                thisFilterGroup.push(['in', ['get', property], item]);
+              }
+            });
+            acc.push(thisFilterGroup);
+          }
+          return acc;
+        },
+        [] as any[]
+      );
+
+      map.setFilter('vacant_properties_tiles_points', ['all', ...mapFilter]);
+      map.setFilter('vacant_properties_tiles_polygons', ['all', ...mapFilter]);
+    } else {
+      console.warn('Layers not found, skipping filter application.');
+    }
+  });
   // map load
   return (
     <div className="customized-map relative max-sm:min-h-[calc(100svh-100px)] max-sm:max-h-[calc(100svh-100px) h-full overflow-auto w-full">
       <Map
         mapLib={maplibregl as any}
         initialViewState={initialViewState}
-        mapStyle={`https://api.maptiler.com/maps/dataviz/style.json?key=${maptilerApiKey}`}
+        mapStyle={mapStyles[currentStyle]?.url}
         onMouseEnter={(e) => changeCursor(e, 'pointer')}
         onMouseLeave={(e) => changeCursor(e, 'default')}
         onClick={onMapClick}
         minZoom={MIN_MAP_ZOOM}
         maxZoom={MAX_MAP_ZOOM}
         interactiveLayerIds={layers}
         onError={(e) => {
-          setHasLoadingError(true);
+          console.log(e);
+          if (
+            e.error.cause ===
+            "The layer 'vacant_properties_tiles_polygons' does not exist in the map's style and cannot be queried for features."
+          )
```

Comment:
Can we handle errors just by calling `setHasLoadingError(true);`?

The `if` statement seems unnecessary, especially because it's comparing `e.error.cause` to a hardcoded error message.